test/htmlrenderer.cpp

```diff
@@ -1621,14 +1621,15 @@ TEST_CASE("Skips contents of <script> tags", "[HtmlRenderer]")
 	std::vector<std::pair<LineType, std::string>> lines;
 	Links links;
 
-	std::ifstream input_file("data/1300-reproducer.html");
-	std::string input;
-	input_file >> input;
```

Comment:
If I use this operator, the test fails.
I didn't look into it further, but I expect this only reads in a single line from the input file.
From a quick check, it looks like this was the only occurrence in our test suite.

I've added the `REQUIRE(lines[0].second == "Visible text");` line to make sure we actually used the full file.

Reply:
It only read the first word, i.e. `<script>`. I don't understand why a single tag could trip up the renderer; perhaps I reproduced the bug manually, fixed it, then wrote a test without checking if it fails without the fix?

Anyway, I see you checked the updated code against 2.21 and it fails there, so everything's well.

Reply:
> perhaps I reproduced the bug manually, fixed it, then wrote a test without checking if it fails without the fix?

No, the test you originally checked in #1302 does reproduce the error.
I used the original version of the test as the basis for the test in my first commit, but had to change it due to [a compilation warning](https://github.com/newsboat/newsboat/runs/24971253710).
It looks like the incorrect line was [introduced to fix that same kind of compilation issue](https://github.com/newsboat/newsboat/commit/6a759bc63db475c59fec2693c4658555f6e6b884#diff-e9f703a8e30d5a4a870c46322a6e2b499e8384a31db74f0557eb6fb63de949c9).